packages/@glimmer/runtime/test/yield-test.ts

```diff
@@ -0,0 +1,145 @@
+import { module, RenderTests, test } from '@glimmer/test-helpers';
+
+class YieldTests extends RenderTests {
+  @test
+  'yield'() {
+    this.render(
+      {
+        layout: '{{#if @predicate}}Yes:{{yield @someValue}}{{else}}No:{{yield to="inverse"}}{{/if}}',
+        args: { predicate: 'activated', someValue: '42' },
+        blockParams: ['result'],
+        template: 'Hello{{result}}{{outer}}',
+        inverse: 'Goodbye{{outer}}'
+      },
+      { activated: true, outer: 'outer' }
+    );
+
+    this.assertComponent('Yes:Hello42outer');
+    this.assertStableRerender();
+  }
+
+  @test({
+    skip: 'glimmer'
+  })
+  'yield to inverse'() {
+    this.render(
+      {
+        layout: '{{#if @predicate}}Yes:{{yield @someValue}}{{else}}No:{{yield to="inverse"}}{{/if}}',
+        args: { predicate: 'activated', someValue: '42' },
+        blockParams: ['result'],
+        template: 'Hello{{result}}{{outer}}',
+        inverse: 'Goodbye{{outer}}'
+      },
+      { activated: false, outer: "outer" }
+    );
+
+    this.assertComponent('No:Goodbyeouter');
+    this.assertStableRerender();
+  }
+
+  @test
+  'yielding to an non-existent block'() {
+    this.render({
+      layout: 'Before-{{yield}}-After'
+    });
+
+    this.assertComponent('Before--After');
+    this.assertStableRerender();
+  }
+
+  @test
+  'yielding a string and rendering its length'() {
+    this.render({
+      layout: `{{yield "foo"}}-{{yield ""}}`,
+      blockParams: ['yielded'],
+      template: '{{yielded}}-{{yielded.length}}'
+    });
+
+    this.assertComponent(`foo-3--0`);
+    this.assertStableRerender();
+  }
+
+  @test({
+    skip: 'glimmer'
+  })
+  'use a non-existent block param'() {
+    this.render({
+      layout: '{{yield someValue}}',
+      args: { someValue: '42' },
+      blockParams: ['val1', 'val2'],
+      template: '{{val1}} - {{val2}}'
+    });
+
+    this.assertComponent('42 - ');
+    this.assertStableRerender();
+  }
+
+  @test
+  'block without properties'() {
+    this.render({
+      layout: 'In layout -- {{yield}}',
+      template: 'In template',
+    });
+
+    this.assertComponent('In layout -- In template');
+    this.assertStableRerender();
+  }
+
+  /*
```

Comment:
Any thoughts on a good way to translate these data driven tests into class-based tests, @chadhietala?